modules/3d/src/octree.cpp

```diff
@@ -344,19 +349,22 @@ bool deletePointRecurse(Ptr<OctreeNode>& _node)
     }
 }
 
-void insertPointRecurse( Ptr<OctreeNode>& _node,  const Point3f& point, int maxDepth)
+bool insertPointRecurse( Ptr<OctreeNode>& _node,  const Point3f& point, int maxDepth)
 {
     OctreeNode& node = *_node;
-    if(!node.isPointInBound(point, node.origin, node.size))
+    if(node.depth==0 && !node.isPointInBound(point, node.origin, node.size))
     {
         CV_Error(Error::StsBadArg, "The point is out of boundary!");
     }
+    else if (!node.isPointInBound(point, node.origin, node.size)){
```

Comment:
I believe we should not call `node.isPointInBound(point, node.origin, node.size)` twice for performance perspective. Cache it result through variable.